### PR TITLE
1340: Correct email domain not always used when adding a contributor to a PR

### PR DESCRIPTION
**Pull request by zhaosongzs**
Now, the email domain for additional contributors is always the default "openjdk.org" domain because we hardcoded it.

To resolve this issue, we should use the domain in jcheck configuration file.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace

### Issue
 * [SKARA-1340](https://bugs.openjdk.org/browse/SKARA-1340): Correct email domain not always used when adding a contributor to a PR

### Reviewers
 * [Guoxiong Li](https://openjdk.org/census#gli) (@lgxbslgx - Committer) ⚠️ Review applies to [b2bc9c37](https://git.openjdk.org/skara/pull/1513/files/b2bc9c375fe7f27dba51255bd48919397cdf9717)
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to [bf623cfa](https://git.openjdk.org/skara/pull/1513/files/bf623cfa0201ecf74fa9d2e3cf42335269518a3d)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/skara.git pull/1513/head:pull/1513` \
`$ git checkout pull/1513`

Update a local copy of the PR: \
`$ git checkout pull/1513` \
`$ git pull https://git.openjdk.org/skara.git pull/1513/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1513`

View PR using the GUI difftool: \
`$ git pr show -t 1513`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/skara/pull/1513.diff">https://git.openjdk.org/skara/pull/1513.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/skara/pull/1513#issuecomment-1536591219)

### CODE REVIEW

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/ContributorCommand.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ private Optional<EmailAddress> parseUser(String user, PullRequest pr, CensusInst
         }
 
         if (contributor.fullName().isPresent()) {
-            return Optional.of(EmailAddress.from(contributor.fullName().get(), contributor.username() + "@openjdk.org"));
+            return Optional.of(EmailAddress.from(contributor.fullName().get(), contributor.username() + "@" + censusInstance.configuration.census().domain()));
```
Comment (lgxbslgx) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/ContributorCommand.java:90]: The `configuration` is a private field in class `CensusInstance`, so we should use the getter method `configuration()`. But it is curious that the building and testing of the `Pre-submit tests` succeeded.

**Comment by zhaosongzs** [comment]
> Looks good.
> 
> > But it is curious that the building and testing of the Pre-submit tests succeeded.
> 
> But still strange.

The field is not private, it's protected

**Comment by lgxbslgx** [comment]
> The field is not private, it's protected. And `LimitedCensusInstance` and `ContributorCommand` are both under `org.openjdk.skara.bots.pr`

I know why I misunderstand it. I mis-read the `CensusInstance` class of the package `org.openjdk.skara.bots.mlbridge` by using the GitHub symbol search feature. So it is also good to use the field directly. Nevermind.

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/ContributorCommand.java [hunk 2]
```diff
@@ -87,7 +87,7 @@ private Optional<EmailAddress> parseUser(String user, PullRequest pr, CensusInst
         }
 
         if (contributor.fullName().isPresent()) {
-            return Optional.of(EmailAddress.from(contributor.fullName().get(), contributor.username() + "@openjdk.org"));
+            return Optional.of(EmailAddress.from(contributor.fullName().get(), contributor.username() + "@" + censusInstance.configuration().census().domain()));
```
Comment (erikj79) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/ContributorCommand.java:90]: Could do with a line break here to make it more readable, especially in diffs.

**Review by erikj79 — APPROVED** [review]

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/ContributorCommand.java (hunk 2 above)
Comment (zhaosongzs) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/ContributorCommand.java:90]: Sure

**Pull request closed without merging**